src/pylexibank/cldf.py

```diff
@@ -102,6 +103,104 @@ def cognate_id(self, kw):
     def tokenize(self, item, string):
         if self.dataset.tokenizer:
             return self.dataset.tokenizer(item, string)
+    
+    def add_segments(self, **kw):
+        """
+        :return: dict with the newly created lexeme
+        """
+        kw_ = kw.copy()
+
+        # Do we have morpheme segmentation on top of phonemes?
+        with_morphemes = '+' in self['FormTable', 'Segments'].separator
+
+        language, concept, value, form, segments = [kw_.get(v) for v in [
+            'Language_ID', 'Parameter_ID', 'Value', 'Form', 'Segments']]
+        # check for required kws
+        if language is None or concept is None or value is None \
+                or form is None or segments is None:
+            raise ValueError('language, concept, value, form, ' 
+                    'and segments must be supplied')
+        
```

Comment:
Why don't we just make the function signature explicit?: 

def add_segments(self, language, concept, value, form, segments):

Reply:
Yes, but we didn't have that in self.add_lexemes, where it would also be in order (as this test is useful, right?), and if we change add_lexemes, we'll loose backwards compat. The disadvantage would mostly be mnemotechnics: you'd have to remember your order. Another disadvantage: I'd have to add the four items in another statement to the `**kw`` anyway, and I'd also have to check for `None` (which can also happen, as you may accidentally retrieve a None value in lingpy, for example.

Reply:
But `add_segments` does something different, right? It doesn't add complete rows of CLDF tables, but rather a single column value, correct? If so, having a different signature would be ok, I think.

Reply:
it adds a cldf row. but it does not split a value into forms, and it does not tr yto segment the stuff.

Reply:
Ok, I see. But then isn't this only a special case of `add_form`? I also regret the name `add_lexemes` a bit, when it actually adds rows to the `FormTable`, but then we should either rename `add_lexemes` or `add_form` - because clearly one is a special case of the other (and should possibly even be called by the other) and consistency would be important.

Reply:
Hierarchy is:

* `add_segments`: complete user-controled, all manual 
* `add_form`: user splits values, but has us segment the data, or refuses to add segments
* `add_lexemes`: we do almost everything, if orthoprofile, etc. is there

I was first thinking of actually modifying `add_lexemes` directly (but this would cost backwards compat, I am afraid), and have something like: if form is given, don't split the form, if segments are given and form is given, don't segmentize. We in fact have this statement for segments in `add_lexemes` but not for the form, only for the segments. Yet I thought that it would generally be better for educative reasons to use indeed three functions. If one calls another one is in fact all the same to me, and would be better, but three functions force the user to think what they have. And that nobody knew for sure WHAT add_lexemes does, was obvious, also from the long discussions we had on several datasets in the past.

Reply:
I'd opt for renaming add_lexemes to add_value, or make an add_value function for now, deprecate add_lexemes, and maybe even have the general splitting properties been passed to add_value (!), to avoid the overriding of internal attributes, etc., which again led to an incredible confusion among lexibankers, which we still see in the current datasets.

Reply:
In fact, that would be the smoothest transition, since it would not require to really start everything again, but we'd step by step change from add_lexemes to add_value, and we can internally regularize the code to decrease redundancy.

Reply:
Ok, I see. But if we go this way, then I'd strongly argue for more consistent names:
- `add_forms` (renamed from `add_lexemes`): We have potentially mulltiple forms, let pylexibank do the splitting
- `add_form`: the generic "adds one row in FormTable" method - called by all of the others - maybe an internal method (`_add_form`) if we really want to force users to chose a specific one.
- `add_form_with_segments`: renamed from `add_segments`.

Reply:
Agreed. Only for `add_forms` I'd suggest: `add_forms_from_value`, as this is more explicit even with respect to what it does, right? I can try to add those, leaving `add_lexemes` untouched, and we can see how long we need for the transition, giving this a version 1.1.2 or similar?

Reply:
ok, I like `add_forms_from_value`, too. Why would you leave `add_lexemes` untouched rather than refactor it calling `add_form` once we have this? From my experience, providing opportunity for copies of code to get out of sync is a bad idea.

Reply:
Okay, I'll try and quickly work on something. I may need help or somebody else to take over in fully refactoring, if this is okay.

Reply:
I think add_forms() calling add_form() makes a lot of sense. A little reluctant about add_forms_from_value (it's ugly) but I think we've learnt that we need to make things more explicit here.

Reply:
Don't want to bikeshed here, but flags might also help e.g. 

```
ds.add_form(form, segmentable=False)
```

Reply:
(deleted my comments here -- I see that you've got it sorted out below) 👍